source/agora/common/Transaction.d

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+
+    Defines the data structure of a transaction
+
+    The current design is heavily influenced by Bitcoin: as we have an UTXO,
+    starting with a simple input/output approach seems to make the most sense.
+    Script is not implemented: instead, we currently only do a simple signature
+    verification.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Transaction;
+
+import agora.common.Data;
+import agora.common.crypto.Key;
+
+
+/*******************************************************************************
+
+    Currency amount to use
+
+    This is simply an alias to `long` at the moment, and should be made smarter.
+    Currently this has two major drawbacks:
+    - It does not do any overflow / underflow checks (over the total amount)
+    - It can go negative
+    We should probably wrap this in a smarter type to do currency operations,
+    and move it to a shared location, as currency operations might be performed
+    independently of `Transaction`.
+
+*******************************************************************************/
+
+public alias Amount = long;
+
+/*******************************************************************************
+
+    Represents a transaction (shortened as 'tx')
+
+    Agora uses an UTXO model for transactions, so the transaction format is
+    originally derived from that of Bitcoin.
+
+*******************************************************************************/
+
+public struct Transaction
+{
+    /// The list of `outputs` from previous transaction(s) being spent
+    Input[] inputs;
```

Comment:
```suggestion
    Input[] inputs;
    
```

I know we don't have a style guide, but I find it hard to look a fields when they're so tightly packed..

Reply:
I don't mind packed when things are self explanatory, but I don't mind the other way either.

Reply:
I'm ok with packing if there's no comments in-between. Because the comments and the fields they comment should be grouped together. And you do that with spacing.

```
/// comment1
int field1;

/// comment2
float field2;
```

But I'm totally ok with this:

```
int field1;  /// hello
float field2;  /// hello 2
```